authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.authorization.utils;
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AlternativeAuthorizationResourceActionProvider;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.service.CommonPermissionCheckingUtils;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+
+@Component
+public class EventAuthorizationUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventAuthorizationUtils.class);
+
+    private AlternativeAuthorizationResourceActionProvider actionAlternativeProvider;
+
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
+    public EventAuthorizationUtils(CommonPermissionCheckingUtils permissionCheckingUtils,
+            AlternativeAuthorizationResourceActionProvider actionAlternativeProvider) {
+        this.actionAlternativeProvider = actionAlternativeProvider;
+        this.permissionCheckingUtils = permissionCheckingUtils;
+    }
+
+    public void checkPermissionBasedOnResourceTypeAndCrn(Collection<EventAuthorizationDto> eventAuthorizationDtos) {
+        throwIfNull(eventAuthorizationDtos,
```

Comment:
Isn't guava's `Preconditions.checkNotNull()` good enough for you?

Reply:
As I can see - but correct me if I'm wrong -, this `checkNotNull` method throws NPE which is exactly that what I would like to avoid because we're not trying to access some data that is not there, but we're doing a preliminary check whether the caller - method - has passed a proper object or not. If not, then throwing an `IllegalArgumentException` is much more descriptive, for me at least. (just think about the name itself of the exception. It is quite self-explanatory)